tests/javascript/Tiltfile

```diff
@@ -0,0 +1,59 @@
+# Some common test configurations to easily get up and running with Tilt for Tests
+# (see docs: https://docs.tilt.dev/tests_in_tilt.html)
+
+
+def _test_js(name, pkmanager, dir, deps=None, with_install=False, project_root='', ignore=None, extra_args=None):
+    if not project_root:
+        project_root = dir
+
+    install_deps = deps_for_package_manager(pkmanager, project_root)
+    resource_deps = []
+    if with_install:
+        install_resource = pkmanager + '-install'
+        install_cmd = 'cd {dir} && {pkmanager} install'.format(
+            dir=dir, pkmanager=pkmanager)
+        local_resource(install_resource, install_cmd, install_deps)
+        resource_deps = [install_resource]
+
+        # double check that we could find package.json etc. correctly
+        for f in install_deps:
+            if not os.path.exists(f):
+                fail('''Expected file "{}" not found, refusing to attempt to install JS deps.
```

Comment:
@landism better? (i'm a liiiittle worried about what will happen if you're trying to configure a project that doesn't have a lockfile yet, or has a package/lockfile of a different name than we expect...? but we'll cross that bridge when we come to it.)

Reply:
Ach, sorry, for some reason I was thinking we only did this when project_root was specified (maybe the `project_root = dir` is a bit confusing as a reader?). It seemed like a much clearer case for error when I thought this meant they were specifying a project_root that didn't contain the project files.

Sigh, yeah, a warn is probably best. Sorry for the churn.

Reply:
🙃